week9/algoorgoal/src/components/common/Text.tsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import styled from 'styled-components';
+import { ColorType, SizeType, ThemeType } from '../../common/styles/theme';
+
+const StyledText = styled.span<{
+  color: ColorType;
+  size: SizeType;
+  theme: ThemeType;
+}>`
+  color: ${(props) => props.theme.color[props.color]};
+  font-size: ${(props) => props.theme.size[props.size]};
+`;
+
+interface TextPropsType {
+  color: ColorType;
+  size: SizeType;
+  children: React.ReactNode;
+  as?: string;
+}
+
+export default function Text({ children, ...props }: TextPropsType) {
+  return <StyledText {...props}>{children}</StyledText>;
+}
```

Comment:
color나 size 같은 경우는 많이 쓰는 기본값이 있으니까 optional로 만들고 이렇게 쓰는게 좋습니다.

```suggestion
import React from 'react';
import styled from 'styled-components';
import { ColorType, SizeType, ThemeType } from '../../common/styles/theme';

const StyledText = styled.span<{
  color: ColorType;
  size: SizeType;
  theme: ThemeType;
}>`
  color: ${(props) => props.theme.color[props.color]};
  font-size: ${(props) => props.theme.size[props.size]};
`;

interface TextPropsType {
  color?: ColorType;
  size?: SizeType;
  children: React.ReactNode;
  as?: string;
}

export default function Text({ children, size = 'medium', color='primary'}: TextPropsType) {
  return <StyledText size={size} color={color}>{children}</StyledText>;
}
```

Reply:
오 감사합니다! 수정해보겠습니다